book/src/design/circuit/gadgets/sinsemilla/note-commit.md

```diff
@@ -0,0 +1,374 @@
+# NoteCommit
+
+## Message decomposition
+
+$\SinsemillaCommit$ is used in the
+[$\NoteCommit$ function](https://zips.z.cash/protocol/protocol.pdf#concretesinsemillacommit).
+The input to $\SinsemillaCommit$ is:
+
+$$\DiversifiedTransmitBaseRepr \bconcat
+  \DiversifiedTransmitPublicRepr \bconcat
+  \ItoLEBSP{64}(\mathsf{v}) \bconcat
+  \ItoLEBSP{\BaseLength{Orchard}}(\rho) \bconcat
+  \ItoLEBSP{\BaseLength{Orchard}}(\psi),$$
+
+where:
+- $\DiversifiedTransmitBaseRepr, \DiversifiedTransmitPublicRepr$ are representations of
+  Pallas curve points, with $255$ bits used for the $x$-coordinate and $1$ bit used for
+  the $y$-coordinate.
+- $\rho, \psi$ are Pallas base field elements.
+- $\mathsf{v}$ is a $64$-bit value.
+- $\BaseLength{Orchard} = 255.$
+
+Sinsemilla operates on multiples of 10 bits, so we start by decomposing the message into
+chunks:
+
+$$
+\begin{aligned}
+\DiversifiedTransmitBaseRepr &= a \bconcat b_0 \bconcat b_1 \bconcat b_2 \\
+ &= (\text{bits 0..=249 of } \mathsf{x(g_d)}) \bconcat
+    (\text{bits 250..=253 of } \mathsf{x(g_d)}) \bconcat
+    (\text{bit 254 of } \mathsf{x(g_d)}) \bconcat
+    (ỹ \text{ bit of } \mathsf{g_d}) \\
```

Comment:
I cannot find any constraints (in this PR or implemented in https://github.com/zcash/orchard/pull/101) that check this bit matches the LSB of y(g<sub>d</sub>).

Reply:
This has now been documented in 69c057107d3657ddfed383b26193eb7c80a0dcc7 and implemented in 3f711407733cab65de1e145ed1f6083fe0215d45.